lib/chef/search/query.rb

```diff
@@ -89,7 +89,7 @@ def search(type, query='*:*', *args, &block)
         if block
           response["rows"].each { |row| block.call(row) if row }
           unless (response["start"] + response["rows"].length) >= response["total"]
```

Comment:
Is it possible to have the following condition: ?
unless ((response["start"] + response["rows"].length) >= response["total"]) || (response["rows"].length == 0)
Any background issues? 
In case if index is corrupted on chef-server (returns not correct total) It will help.
Thanks.